lib/src/barback.dart

```diff
@@ -34,7 +34,7 @@ import 'package:pub_semver/pub_semver.dart';
 /// constraint and barback's version.
 ///
 /// [compat]: https://gist.github.com/nex3/10942218
-final pubConstraints = {
+final pubConstraints = <String, VersionRange>{
```

Comment:
There's no guarantee that `new VersionConstraint.parse()` will emit a `VersionRange`, so this annotation will probably end up generating an implicit cast warning at some point. If we rely on these being `VersionRange`s, you should explicitly cast them with `as`.